### PR TITLE
Workaround self.version issue with composer 2.2+.

### PR DESCRIPTION
**Pull request by kporras07**

### CODE REVIEW

File: src/Commands/BuildToolsBase.php
Comment (jabxjab) [inline src/Commands/BuildToolsBase.php]: @kporras07 `if` statement is redundant

File: src/Commands/BuildToolsBase.php
Comment (jabxjab) [inline src/Commands/BuildToolsBase.php]: @kporras07 why not `$this->passthru($command);`?

File: src/Commands/BuildToolsBase.php
Comment (jabxjab) [inline src/Commands/BuildToolsBase.php]: @kporras07 how about introducing  a unified variable for `create-project` command additional options, ie. `$create_project_options`?
```php
// ...
$source_project = $source;
$additional_commands = [];
$create_project_options = [];
// ...
if ('' === $stability) {
    if ($source === 'git@github.com:pantheon-upstreams/drupal-recommended.git') {
        $create_project_options[] = '--stability dev';
        // ...
    }
} else {
    $create_project_options[] = "--stability $stability";
}
// ...
$create_project_options[] = '--no-install';
// ...
$create_project_command = sprintf(
    'composer create-project --working-dir=%s %s %s %s -n %s',
    $tmpsitedir, 
    $repository, 
    $source_project,
    $target,
    implode(' ', $create_project_options)
);
// ...
```

Comment (kporras07) [reply src/Commands/BuildToolsBase.php]: Fixed. Thanks!

**Comment by kporras07** [comment]
test_drupal_9_github has been failing since a while ago. I'll review these tests in a future PR so once approved, I'd like to merge this even if that test is failing

**Review by greg-1-anderson — APPROVED** [review]

**Pull request merged**
